docs/how_to/use_optimizationlevel.rst

```diff
@@ -0,0 +1,336 @@
+represented by the ``['id', 'rz', 'sx', 'x', 'cx', 'reset']``.
+
+.. testcode::
+
+    qc_b0 = transpile(qc,backend=backend,optimization_level=0)
+    print(qc_b0)
+
+
+.. testoutput::
+
+    global phase: 5π/8
+                                                                               »
+         q0_0 -> 0 ────────────────────────────────────────────────────────────»
+                   ┌───┐                                                  ┌───┐»
+         q0_1 -> 1 ┤ X ├──────────────────────────────■────────────────■──┤ X ├»
+                   └─┬─┘┌─────────┐┌────┐┌─────────┐┌─┴─┐┌──────────┐┌─┴─┐└─┬─┘»
+         q0_2 -> 2 ──■──┤ Rz(π/2) ├┤ √X ├┤ Rz(π/2) ├┤ X ├┤ Rz(-π/4) ├┤ X ├──■──»
+                        └─────────┘└────┘└─────────┘└───┘└──────────┘└───┘     »
+    ancilla_0 -> 3 ────────────────────────────────────────────────────────────»
+                                                                               »
+    ancilla_1 -> 4 ────────────────────────────────────────────────────────────»
+                                                                               »
+    «                                                                         »
+    «     q0_0 -> 0 ───────■────────────────────────────────■─────────────────»
+    «                    ┌─┴─┐┌─────────┐┌───┐┌──────────┐┌─┴─┐     ┌───┐     »
+    «     q0_1 -> 1 ──■──┤ X ├┤ Rz(π/4) ├┤ X ├┤ Rz(-π/4) ├┤ X ├──■──┤ X ├──■──»
+    «               ┌─┴─┐└───┘└─────────┘└─┬─┘├─────────┬┘└───┘┌─┴─┐└─┬─┘┌─┴─┐»
+    «     q0_2 -> 2 ┤ X ├──────────────────■──┤ Rz(π/4) ├──────┤ X ├──■──┤ X ├»
+    «               └───┘                     └─────────┘      └───┘     └───┘»
+    «ancilla_0 -> 3 ──────────────────────────────────────────────────────────»
+    «                                                                         »
+    «ancilla_1 -> 4 ──────────────────────────────────────────────────────────»
+    «                                                                         »
+    «                          ┌─────────┐                       
+    «     q0_0 -> 0 ─────■─────┤ Rz(π/4) ├───■───────────────────
+    «                  ┌─┴─┐   ├─────────┴┐┌─┴─┐            ┌───┐
+    «     q0_1 -> 1 ───┤ X ├───┤ Rz(-π/4) ├┤ X ├────────────┤ X ├
+    «               ┌──┴───┴──┐├─────────┬┘├───┴┐┌─────────┐└─┬─┘
+    «     q0_2 -> 2 ┤ Rz(π/4) ├┤ Rz(π/2) ├─┤ √X ├┤ Rz(π/2) ├──■──
+    «               └─────────┘└─────────┘ └────┘└─────────┘     
+    «ancilla_0 -> 3 ─────────────────────────────────────────────
+    «                                                            
+    «ancilla_1 -> 4 ─────────────────────────────────────────────
+    «                                                            
+
+When you set the :attr:`~qiskit.transpile.optimization_level` to 1,the circuit undergoes a light optimization process that focuses on collapsing adjacent gates, 
```

Comment:
I understand the idea and the reason you gave me, the problem is a good idea using those methods or explain in a general way, because how to use optimization_level and not the theoretical part? I'll proposal a general way to explain all of this